sklearn/preprocessing/_encoders.py

```diff
@@ -604,6 +608,18 @@ def fit(self, X, y=None):
         self
 
         """
+
+        checkd_X = check_array(X, dtype=None)
+        contain_str = checkd_X.dtype == object or np.issubdtype(
```

Comment:
object dtype does not entail strings I think

Reply:
I am not sure. I saw few tests, which consists of examples like np.array([['strings1','strings2']], np.object). Please let me know, if can remove it.

Reply:
object dtype is a good clue, but not sufficient. might be better to check `type(checkd_X[0])`